Documentation/CACHE-HOWTO.md

```diff
@@ -0,0 +1,204 @@
+<!---
+ Licensed to the Apache Software Foundation (ASF) under one or more
+ contributor license agreements.  See the NOTICE file distributed with
+ this work for additional information regarding copyright ownership.
+ The ASF licenses this file to You under the Apache License, Version 2.0
+ (the "License"); you may not use this file except in compliance with
+ the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+### Overview
+
+Cache configuration provides you additional control over incremental maven behavior. Follow it step by step to
+understand how it works and figure out your optimal config
+
+### Minimal config
+
+Absolutely minimal config which enables incremental maven with local cache
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<cache xmlns="org:apache:maven:cache:config:v1"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="org:apache:maven:cache:config:v1 cache-config.xsd">
+
+    <configuration>
+        <enabled>true</enabled>
+        <hashAlgorithm>XX</hashAlgorithm>
```

Comment:
replace XX with SHA-256 will ease people just copy/pasting

Reply:
it's for xxHash which we found as a most fastest

Reply:
Note that xxHash by Yann Collet is non-cryptographic. I don't know whether it makes a difference here. I consider all SHA hashes as abysmally slow compared to xxHash and BLAKE3.

Reply:
No cryptographic strength is required here, in terms of quality xxHash has Score 10 in SMHasher test suite and [no reported problems](https://github.com/rurban/smhasher).
@hboutemy SHA-1, SHA-256, SHA-384, SHA-512 is also supported, just to mention that on a large project you can save several seconds just by using fast hash function.

Reply:
I have noticed this on Resolver with SHA-256 and SHA-512. Horribly slow.

Reply:
![benchmark](https://habrastorage.org/webt/iy/o5/zj/iyo5zjj66fer1yymqvg-acx9aum.png)
Setup:
- run on Oracle JDK 1.8.0.201 on Windows and Linux
- use 100 jar files with different size from 3 KB to 90 MB, total size 474 MB
- read in single thread, calculate hash for each file and total checksum
- separate folder for each algorithm to avoid disk cache testing
- 100 rounds to minimize CPU interference from external loads
- time in seconds, candle body is 25-75 percentile, shadow is min/max values

Results:
- xxHash shows stable time (3-4 sec) on both Windows and Linux platforms
- xxHash with Memory Mapped buffer (XXMM) gives another small gain on Linux, useful on CI
- SHA-256 about 4 times slower with a difference of 10 seconds even on this small test

P.S. also tested 5157 files with total size 2.6 GB, difference between xxHash and SHA-256 was about 1.5-2 min

Reply:
thank you all for the detailed explanation: I learned something new and interesting today :)